server/embed/config.go

```diff
@@ -628,6 +628,9 @@ func (cfg *configYAML) configFromFile(path string) error {
 	if cfg.ClusterState == "" {
 		cfg.ClusterState = ClusterStateFlagNew
 	}
+	if cfg.AutoCompactionMode == "" {
```

Comment:
It seems it is trying to align with commandline. https://github.com/etcd-io/etcd/blob/5baf8377bbb72d1c7f2b370fdb921091a069ab9d/server/etcdmain/config.go#L238

But etcd ignores any config from commandline or env if you use `config-file`.

https://github.com/etcd-io/etcd/blob/5baf8377bbb72d1c7f2b370fdb921091a069ab9d/server/etcdmain/help.go#L40)

IMO, I think the user should set the auto compaction mode explicitly in config-file since it is easy to add and maintain options in yaml than commandline. And it doesn't depend on the default behaviour. 

And I also think we can add the validation in `parseCompactionRetention`. https://github.com/etcd-io/etcd/blob/ddd65e7fe2e2d6fbb9a89a07ddbef79412885659/server/embed/etcd.go#L794-L802

If the `retention` is just number and `mode` is empty, we should return error.

Reply:
Thanks for your review. 

Make sense. Currently if auto-compaction-mode is set as '1m' in config-file, etcd will report "unsupported compaction mode", but set number "1", will disable auto compaction silently, it's confusion.